src/core/lib/security/security_connector/local_security_connector.cc

```diff
@@ -0,0 +1,256 @@
+    grpc_handshake_manager* handshake_manager) {
+  tsi_handshaker* handshaker = nullptr;
+  GPR_ASSERT(local_tsi_handshaker_create(nullptr, false /* is_client */,
+                                         &handshaker) == TSI_OK);
+  grpc_handshake_manager_add(handshake_manager, grpc_security_handshaker_create(
+                                                    handshaker, &sc->base));
+}
+
+static int local_channel_cmp(grpc_security_connector* sc1,
+                             grpc_security_connector* sc2) {
+  grpc_local_channel_security_connector* c1 =
+      reinterpret_cast<grpc_local_channel_security_connector*>(sc1);
+  grpc_local_channel_security_connector* c2 =
+      reinterpret_cast<grpc_local_channel_security_connector*>(sc2);
+  int c = grpc_channel_security_connector_cmp(&c1->base, &c2->base);
+  if (c != 0) return c;
+  return strcmp(c1->target_name, c2->target_name);
+}
+
+static int local_server_cmp(grpc_security_connector* sc1,
+                            grpc_security_connector* sc2) {
+  grpc_local_server_security_connector* c1 =
+      reinterpret_cast<grpc_local_server_security_connector*>(sc1);
+  grpc_local_server_security_connector* c2 =
+      reinterpret_cast<grpc_local_server_security_connector*>(sc2);
+  return grpc_server_security_connector_cmp(&c1->base, &c2->base);
+}
+
+static grpc_security_status local_auth_context_from_tsi_peer(
+    tsi_peer* peer, grpc_auth_context** ctx) {
+  if (peer == nullptr || ctx == nullptr) {
+    gpr_log(GPR_ERROR,
+            "Invalid arguments to local_auth_context_from_tsi_peer()");
+    return GRPC_SECURITY_ERROR;
+  }
+  /* Create auth context. */
+  *ctx = grpc_auth_context_create(nullptr);
+  grpc_auth_context_add_cstring_property(
+      *ctx, GRPC_TRANSPORT_SECURITY_TYPE_PROPERTY_NAME,
+      GRPC_LOCAL_TRANSPORT_SECURITY_TYPE);
+  GPR_ASSERT(peer->property_count == TSI_LOCAL_NUM_OF_PEER_PROPERTIES);
+  const tsi_peer_property* tsi_prop = &peer->properties[0];
+  /* Add peer process ID to auth context. */
+  if (strcmp(tsi_prop->name, TSI_LOCAL_PROCESS_ID_PEER_PROPERTY) == 0) {
+    grpc_auth_context_add_property(*ctx, TSI_LOCAL_PROCESS_ID_PEER_PROPERTY,
+                                   tsi_prop->value.data,
+                                   tsi_prop->value.length);
+    GPR_ASSERT(grpc_auth_context_set_peer_identity_property_name(
+                   *ctx, TSI_LOCAL_PROCESS_ID_PEER_PROPERTY) == 1);
+  }
+  if (!grpc_auth_context_peer_is_authenticated(*ctx)) {
+    gpr_log(GPR_ERROR, "Invalid unauthenticated peer.");
+    GRPC_AUTH_CONTEXT_UNREF(*ctx, "test");
+    *ctx = nullptr;
+    return GRPC_SECURITY_ERROR;
+  }
+  return GRPC_SECURITY_OK;
+}
+
+static void local_check_peer(grpc_security_connector* sc, tsi_peer peer,
+                             grpc_auth_context** auth_context,
+                             grpc_closure* on_peer_checked) {
+  grpc_security_status status;
+  status = local_auth_context_from_tsi_peer(&peer, auth_context);
```

Comment:
Do we really need copy peer identity from tsi_peer to auth_context? The peer identity in tsi_peer is just dummy information.

Cann't we just have a function local_auth_context_create(auth_contex), in which we just add a dummy peer identity into auth_context directly.

If so, we do not put anything in tsi_peer. We may even get rid of local_transport_security.

Reply:
Good point. yes, I think we can get rid of tsi_handshaker_result APIs.